kernel/pci/src/lib.rs

```diff
@@ -1,5 +1,20 @@
 //! PCI Configuration Space Access
 //!
+//! ## Terminology
+//!
+//! This crate deals with multiple types of interrupts:
+//! * Legacy (INTx) interrupts are the oldest PCI interrupt representation.
+//!   Four interrupt pins are shared among all devices.
+//!   This crate refers to that by "intx".
+//!
+//! * MSI (Message Signaled Interrupts) appeared with PCI express.
+//!   They allow devices to allocate up to 32 interrupt numbers.
+//!   This crate refers to that by "msi".
+//!
+//! * MSI-X messages appeared with PCIe 3.0.
+//!   They allow devices to allocate up to 2048 interrupt numbers.
+//!   This crate refers to that by "msix".
```

Comment:
formatting & clarifications
```suggestion
//! This crate deals with multiple types of interrupts:
//! * Legacy (INTx) interrupts are the oldest PCI interrupt representation.
//!   * Four interrupt pins are shared among all devices.
//!   * This crate refers to these legacy interrupts as "intx".
//!
//! * MSI (Message Signaled Interrupts) appeared with PCI express.
//!   * They allow devices to allocate up to 32 interrupt numbers.
//!   * This crate refers to these interrupts as "msi".
//!
//! * MSI-X messages appeared with PCIe 3.0.
//!   * They allow devices to allocate up to 2048 interrupt numbers.
//!   * This crate refers to these interrupts as "msix".
```